src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -13,22 +13,24 @@ internal abstract partial class AnalyzerDriver : IDisposable
     {
         internal class CompilationData
         {
-            private readonly Dictionary<SyntaxReference, DeclarationAnalysisData> _declarationAnalysisDataMap;
+            private readonly record struct SyntaxReferenceAndSymbol(SyntaxReference Declaration, ISymbol Symbol);
+            private readonly Dictionary<(ISymbol, int), DeclarationAnalysisData> _declarationAnalysisDataMap;
```

Comment:
>(ISymbol, int) [](http://example.com/codeflow?start=40&length=14)

Consider providing names for the properties to make their purpose clear.

Reply:
Sure, thanks